### ISSUE TITLE
MD045 shown when image has alt text in the optional title

### ISSUE THREAD

**Issue by Licantrop0** [DavidAnson/markdownlint#1236]
[MD045 - Images should have alternate text (alt text)](../blob/v0.34.0/doc/md045.md) is shown when the optional title in the reference syntax is used but the Alternate Text is empty.

Example:
```
![][ref]
...
[ref]: image.jpg "Alternate text"
```

The title text in the reference should be enough for not being flagged as violation, as the Assistive Technologies can read the title instead of the empty alt text, as you can see from this sample:

**In-line**: ![test inline](https://github.com/DavidAnson/markdownlint/assets/2388430/7f548680-c36c-4d0d-ad42-e3794d47d984)
**Reference**: ![xxx][ref]

[ref]: https://github.com/DavidAnson/markdownlint/assets/2388430/7f548680-c36c-4d0d-ad42-e3794d47d984 "test reference"

**Comment by DavidAnson** [comment]
Per mdn, "The title attribute is not an acceptable substitute for the alt attribute.":

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#the_title_attribute

Your example renders a blank "alt" attribute, so the rule is correct to report a violation:

https://markdown-it.github.io/#md3=%7B%22source%22%3A%22!%5B%5D%5Bref%5D%5Cn%5Cn%5Bref%5D%3A%20image.jpg%20%5C%22Alternate%20text%5C%22%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22src%22%7D%7D

**Comment by Licantrop0** [comment]
ah, that's a pity... Thanks for the explanation, I'll close this issue.
I wish there were a way to specify the alt-text in reference rather than in the individual tags, especially when you have to refer to the same image multiple times.

**Comment by DavidAnson** [comment]
If you have control over the Markdown renderer, I think it would be a very reasonable modification to have it substitute title for alternate text. Maybe some renderers already do that? But it's not part of the standard, so this rule does not assume that is being done.
